uint16[4] channels
